### ISSUE TITLE
post_level where clause

### ISSUE THREAD

**Issue by erikjoling** [htmlburger/carbon-fields#383]
Hi, 

With the following code I'd expect the container to show on first-level pages, but the metabox isn't shown. Any ideas why? I'm using carbon fields 2.1.0

```
Container::make( 'post_meta', 'Page Header' )
    ->where( 'post_type', 'IN', array('page') )
    ->where( 'post_level', '=', 1 )
    ->add_fields( array(
        Field::make( 'image', '_header_image_id' ),
        Field::make( 'text', '_header_text' ),
    ) );
```

And when trying to use a custom callable I get an error. Am I doing something wrong?

```
Container::make( 'post_meta', 'Page Header' )
    ->where( 'post_type', 'IN', array('page') )
    ->where( 'post_level', 'CUSTOM', function($post_level) {
        return true;
    })
    ->add_fields( array(
        Field::make( 'image', '_header_image_id' ),
        Field::make( 'text', '_header_text' ),
    ) );
```

```
[23-Oct-2017 11:29:06 UTC] PHP Notice:  Object of class Closure could not be converted to int in /srv/www/[...]/public_html/wp-content/themes/[...]/includes/vendor/htmlburger/carbon-fields/core/Container/Condition/Post_Level_Condition.php on line 21
[23-Oct-2017 11:29:06 UTC] PHP Fatal error:  Uncaught Carbon_Fields\Exception\Incorrect_Syntax_Exception: Supplied comparison value is not a callable: 1 in /srv/www/[...]/public_html/wp-content/themes/[...]/includes/vendor/htmlburger/carbon-fields/core/Exception/Incorrect_Syntax_Exception.php:18
Stack trace:
#0 /srv/www/[...]/public_html/wp-content/themes/[...]/includes/vendor/htmlburger/carbon-fields/core/Container/Condition/Comparer/Custom_Comparer.php(26): Carbon_Fields\Exception\Incorrect_Syntax_Exception::raise('Supplied compar...')
#1 /srv/www/[...]/public_html/wp-content/themes/[...]/includes/vendor/htmlburger/carbon-fields/core/Container/Condition/Condition.php(91): Carbon_Fields\Container\Condition\Comparer\Custom_Comparer->is_correct(2, 'CUSTOM', 1)
#2 /srv/www/[...]/public_html/wp-content/themes/[...]/includes/vendor/htmlburger/carbon-fields/core/Container/Condition/Post_Level_Condition.php(25): Carbon_Fields\Container\Condition\Condition->compare(2, 'CUSTOM', 1)
 in /srv/www/[...]/public_html/wp-content/themes/[...]/includes/vendor/htmlburger/carbon-fields/core/Exception/Incorrect_Syntax_Exception.php on line 18
```

**Comment by atanas-angelov-dev** [comment]
Hi @erikjoling ,

Could you please test this on the latest beta and let me know if the issue persists? (`composer require htmlburger/carbon-fields:2.1.1-beta.5`)

**Comment by erikjoling** [comment]
Case 1 seems to be fixed (`->where( 'post_level', '=', 1 )`

Case 2 with `->where( 'post_level', 'CUSTOM', function($post_level) { return true; } )` still raises the same error.

**Comment by atanas-angelov-dev** [comment]
Both fixed in 2.1.1 - closing.
